src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -356,17 +356,19 @@ public DocumentContext writingDocument(final boolean metaData) {
 
             // sets the writeLimit based on the safeLength
             openContext(metaData, safeLength);
+
+            // there is nothing to read.
+            wire.bytes().readPosition(wire.bytes().writePosition());
         }
-        // there is nothing to read.
-        wire.bytes().readPosition(wire.bytes().writePosition());
+
         return writeContext;
     }
 
     @Override
     public DocumentContext acquireWritingDocument(boolean metaData) {
         if (!DISABLE_THREAD_SAFETY)
             this.threadSafetyCheck(true);
-        if (wire != null && writeContext.isOpen() && writeContext.chainedElement())
+        if (writeContext.wire != null && writeContext.isOpen() && writeContext.chainedElement())
```

Comment:
This broke the chained methodWriter writes when double buffering. It mean `count` never got to zero so the buffer was never written back.